stock_intercompany_bidirectional/models/stock_picking.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (C) 2023 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class StockPickingIntercompanyBidirectional(models.Model):
+    _inherit = "stock.picking"
+
+    def _create_counterpart_picking(self):
+        """Override of parent method to handle intercompany
+        transactions depending on the operation types in the settings"""
+        active_company = self.env.company
+        inter_company = (
```

Comment:
this is company related partner, maybe the `inter_company_partner` name is better